ui/src/app/guards/workspace-guard.service.ts

```diff
@@ -3,24 +3,24 @@ import {
   ActivatedRouteSnapshot, CanActivate, CanActivateChild, Router,
   RouterStateSnapshot
 } from '@angular/router';
-import {serverConfigStore} from 'app/utils/navigation';
-import {WorkspacesService} from 'generated';
+import {currentWorkspaceStore, serverConfigStore} from 'app/utils/navigation';
+import {WorkspaceAccessLevel} from 'generated/fetch';
 import {Observable} from 'rxjs/Observable';
 
 @Injectable()
 export class WorkspaceGuard implements CanActivate, CanActivateChild {
   constructor(
-    private workspaceService: WorkspacesService,
     private router: Router) {}
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
     if (serverConfigStore.getValue().enableResearchReviewPrompt && route.routeConfig.path === 'data' ||
         route.routeConfig.path === 'notebooks') {
-      this.workspaceService.getWorkspace(route.params.ns, route.params.wsid).subscribe(workspace => {
-        if (workspace.accessLevel === 'OWNER'
-            && workspace.workspace.researchPurpose.needsReviewPrompt) {
+      currentWorkspaceStore.subscribe(workspace => {
```

Comment:
This subscription is never released, so these callbacks will build up indefinitely.

Here, I would suggest `const workspace = await currentWorkspaceStore.first().toPromise();`

`first()` only emits the first workspace in the store